packages/theme/src/cli/services/info.test.ts

```diff
@@ -84,3 +101,54 @@ describe('info', () => {
     })
   })
 })
+
+describe('formatThemeInfo', () => {
+  beforeEach(() => {
+    vi.mocked(formatSection).mockImplementation((key, val) => `${key}: ${val}`)
+  })
+
+  afterEach(() => {
+    // eslint-disable-next-line @shopify/cli/no-vi-manual-mock-clear
+    vi.clearAllMocks()
+  })
+
+  test('formats theme info with environment flag', async () => {
+    // Given
+    const themeInfo = {
+      theme: {
+        id: 1234,
+        name: 'Test Theme',
+        role: 'development',
+        shop: 'test-shop.myshopify.com',
+        editor_url: 'https://test-shop.myshopify.com/editor',
+        preview_url: 'https://test-shop.myshopify.com/preview',
+      },
+    }
+
+    // When
+    const result = await formatThemeInfo(themeInfo, {environment: 'production'})
+
+    // Then
+    expect(result).toEqual({
+      customSections: [
+        {
+          title: 'Theme information',
+          body: [{subdued: 'Environment name: production'}],
+        },
+        {
+          title: '',
+          body: [
+            'id: 1234',
+            'name: Test Theme',
+            'role: development',
+            'shop: test-shop.myshopify.com',
+            'editor_url: https://test-shop.myshopify.com/editor',
+            'preview_url: https://test-shop.myshopify.com/preview',
+          ].join('\n\n'),
+        },
+      ],
+    })
+
+    expect(formatSection).toHaveBeenCalledTimes(6)
```

Comment:
How valuable of an assertion are you thinking this is? Can we skip mocking the function entirely?